Dockerfile

```diff
@@ -0,0 +1,17 @@
+# Build the manager binary
+FROM golang:1.10.3 as builder
```

Comment:
shouldn't this change be in a separate PR/commit?

Reply:
Are you referring to the golang version? This is simply the version used by default in kubebuilder 1.0.5, but we could change it back to what it used to be. I wouldn't think of this change as particularly risky though. But happy to change it if you disagree.

Reply:
in general, seems like there re many changes unrelated to upgrading kubebuilder (the commit/pr mentioned).
It on large commit including some configuration changes, documentation, makefile, dockerfile, vendoring updates, code changes...
it make it a bit harder to follow/review and see history later.